src/client/index.js

```diff
@@ -1,17 +1,30 @@
-/* global Meteor, Mongo */
+/* global Meteor, Mongo, Package */
 
-const client = {};
+let collection = {};
 
 if (typeof Meteor !== 'undefined' && Meteor.isClient) {
-  const JoinCollection = new Mongo.Collection('PublishJoin');
+  collection = new Mongo.Collection('PublishJoin');
 
-  client.get = function get(name) {
-    const join = JoinCollection.findOne({
+  collection.get = function get(name) {
+    const join = this.findOne({
       _id: name,
     });
 
     return join && join.value;
   };
+
+  collection.has = function get(name) {
+    return !!this.findOne({
+      _id: name,
+    });
+  };
+
+  if (Package.templating) {
+    Package.templating.Template.registerHelper('getPublishedJoin', name => collection.get(name));
+    Package.templating.Template.registerHelper('hasPublishedJoin', name => collection.has(name));
+  }
 }
 
+const client = collection;
```

Comment:
That's exactly the way publish-counts does it (see https://github.com/percolatestudio/publish-counts/blob/master/client/publish-counts.js) and I really liked it, because it actually is a collection you are working with. You just extend it by your two methods (`get` and `has`).